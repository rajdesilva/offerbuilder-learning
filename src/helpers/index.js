export * from './api-config';
export * from './history';
