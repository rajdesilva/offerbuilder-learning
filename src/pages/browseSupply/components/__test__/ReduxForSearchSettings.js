export const ReduxForSearchSettings = {
  searchparams: {
    destination: {},
    los: 1,
    distance: 10,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [{}],
      },
    ],
    dateRange: {
      startDate: "05/17/2021",
      endDate: "07/17/2021",
    },

    lcn: false,
    remainingCapitalPool: "",
    sortBy: {
      sortOrder: "ASCENDING",
      sortCriteria: "NAME",
    },
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "PCT",
        name: "PCT",
      },
      {
        id: "PCT",
        name: "ATG",
      },
      {
        id: "DEMO_OFFERBUILDER",
        name: "DEMO_OFFERBUILDER",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
