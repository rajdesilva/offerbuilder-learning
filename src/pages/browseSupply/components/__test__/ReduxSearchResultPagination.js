export const PaginationStore = {
  searchedproperties: {
    loading: false,
    properties: [{}],
    totalMatch: 15,
  },
  searchparams: {
    pageOffset: 0,
    pageSize: 10,
  },
};
