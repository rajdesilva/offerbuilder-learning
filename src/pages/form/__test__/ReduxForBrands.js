export const ReduxForBrands = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
        id: "idb",
        name: "Internal Demo Brand",
      },
    ],
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
