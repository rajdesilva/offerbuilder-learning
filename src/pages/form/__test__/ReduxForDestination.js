export const reduxStore = {
  supplysearchParams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [],
      channels: [],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channels: {
    // loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: null,
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "AT_KINDERHOTELS",
        name: "Ihr Bett im Allg√§u",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
