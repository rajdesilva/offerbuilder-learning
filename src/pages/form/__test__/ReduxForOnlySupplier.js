export const ReduxForOnlySupplier = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [
        {
          id: "ntp",
          name: "NTP",
        },
      ],
      channels: [
        {
          id: "AT_CORDIAL",
          name: "",
        },
      ],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
