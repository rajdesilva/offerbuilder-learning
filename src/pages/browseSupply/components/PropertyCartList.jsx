import React, { Fragment, useMemo, useState } from "react";
import { Row, Rate, Table, Empty, Button, Space, Tooltip } from "antd";
import { useSelector } from "react-redux";
import { RightOutlined, InfoCircleOutlined } from "@ant-design/icons";
import defaultImg from "../../../assets/images/default-img.svg";
import { DeleteCartItem } from "./DeleteCartItem";
import Modal from "antd/lib/modal/Modal";
import { displayMargin, getFormattedValue } from "../../../helpers/utility";
import { FormattedMessage, useIntl } from "react-intl";
import { appConstants } from "../../../common";
import { useRouteMatch } from "react-router-dom";
import styles from './css/PropertyCartList.module.less';

export function PropertyCartList(props) {
  const intl = useIntl();
  const [visible, setVisible] = useState(false);
  const { selectedProperties } = useSelector((state) => ({
    selectedProperties: window.getValue(state, "propertycart.cartItems"),
  }));
  const isNewOfferSettingsPage = useRouteMatch({
    path: "/offers/create-new-offer/3",
    strict: true,
    sensitive: true,
  });
  const columns = useMemo(
    () => [
      {
        title: <FormattedMessage id="nemo.properties" />,
        dataIndex: "",
        className:styles.propertyList,
        width: "12%",
        render: (row) =>
          window.getValue(row, "img") ? (
            <img
              onError={(ev) => (ev.target.src = defaultImg)}
              src={window.getValue(row, "img")}
              alt=""
              className={styles.cartListImg}
              data-testid="image"
            />
          ) : window.getValue(row, "images") ? (
            <img
              onError={(ev) => (ev.target.src = defaultImg)}
              src={window.getValue(row, "images[0]")}
              alt=""
              className={styles.cartListImg}
              data-testid="image"
            />
          ) : (
            <Empty
              data-testid="empty-image"
              image={Empty.PRESENTED_IMAGE_SIMPLE}
              description={<span>No Image </span>}
              className={styles.emptyCartItemImg}
            />
          ),
      },
      {
        className: styles.headingColor,
        dataIndex: "",
        width: "40%",
        render: (text, row) => (
          <div>
            <div className={styles.fw600}>
              {row.hotelName} &nbsp;
              {row.lcn ? (
                <span className={styles.propertyBox} data-testid="lcn">
                  {"LCN"}
                </span>
              ) : (
                <Fragment />
              )}
            </div>
            <div>
              {row.city},&nbsp;
              {row.country} &nbsp;
              {row.rating ? (
                <span>
                  <Rate disabled defaultValue={row.rating} />
                </span>
              ) : (
                <Fragment />
              )}
            </div>
          </div>
        ),
      },
      {
        title: <FormattedMessage id="nemo.margin" />,
        dataIndex: "",
        className: styles.margin,
        width: "10%",
        render: (row) => (
          <Row>
            {displayMargin(row.lowestMargin, row.highestMargin) || (
              <FormattedMessage id="nemo.n/a" />
            )}
          </Row>
        ),
      },
      {
        title: <FormattedMessage id="nemo.supplier" />,
        dataIndex: "",
        className: styles.margin,
        width: "20%",
        render: (row) => (
          <Row>
            {row.supplier ? row.supplier.toUpperCase() : ""}-{row.channel}
          </Row>
        ),
      },
      {
        dataIndex: "",
        title: <FormattedMessage id="nemo.capitalPool" />,
        className: styles.margin,
        width: "13%",
        render: (row) => (
          <Fragment>
            {row.lcn ? (
              <Row>
                {row.remainingCapital
                  ? getFormattedValue(row.remainingCapital)
                  : ""}{" "}
                EUR
              </Row>
            ) : (
              <Fragment />
            )}
          </Fragment>
        ),
      },
      {
        dataIndex: "",
        width: "5%",
        render: (row) => {
          return (
            <Row justify="end">
              <DeleteCartItem row={row} />
            </Row>
          );
        },
      },
    ],
    []
  );

  const showModal = () => {
    setVisible(true);
  };

  const handleCancel = () => {
    setVisible(false);
  };

  const handleProceed = () => {
    if (props.next) {
      props.next();
    }
    setVisible(false);
  };

  return (
    <Fragment>
      <Button
        data-testid="property-list-modal"
        type="primary"
        ghost
        onClick={showModal}
        size="large"
        disabled={!(selectedProperties && selectedProperties.length > 0)}
      >
        <span className={styles.fs14}>
          <FormattedMessage id="nemo.propertiesInOffer" />
        </span>
        &nbsp;
        <span className={styles.cartTotalCount} data-testid="properties-count">
          {selectedProperties ? selectedProperties.length : "0"}
        </span>
      </Button>
      <Modal
        title={<FormattedMessage id="nemo.propertiesInOffer" />}
        visible={visible}
        width={1080}
        centered
        className={styles.cartListModal}
        onCancel={handleCancel}
        data-testid="property-cart-list-modal"
        destroyOnClose
        footer={
          props.isEditFlow
            ? null
            : [
                <Button key="back" onClick={handleCancel} data-testid="cancel">
                  <FormattedMessage id="nemo.cancel" />
                </Button>,
                <Button
                  key="submit"
                  type="primary"
                  onClick={handleProceed}
                  data-testid="proceed"
                  disabled={
                    !(selectedProperties && selectedProperties.length > 0)
                  }
                >
                  {isNewOfferSettingsPage ? (
                    <Fragment>
                      <FormattedMessage id="nemo.proceedToMarketing" />
                      &nbsp;{" "}
                    </Fragment>
                  ) : (
                    <Fragment>
                      <FormattedMessage id="nemo.proceedToSettings" />
                      &nbsp;
                    </Fragment>
                  )}
                  <RightOutlined className={styles.fs10}/>
                </Button>,
              ]
        }
      >
        <Space direction="vertical" size="middle">
          <Space size="middle">
            <div data-testid="properties-selected">
              <FormattedMessage id="nemo.propertiesSelected" />
              &nbsp;
              {selectedProperties ? selectedProperties.length : "0"}/
              {appConstants.MAXIMUM_CART_ITEMS_COUNT}
              &nbsp;
            </div>
            <div>
              <Tooltip
                title={intl.formatMessage(
                  {
                    // Matching ID as above
                    id: "nemo.maximumCartItemCount",
                    // Default Message in English
                    defaultMessage: "{count} properties maximum can be added.",
                  },
                  { count: appConstants.MAXIMUM_CART_ITEMS_COUNT }
                )}
              >
                <InfoCircleOutlined className={styles["fs-16"]} />
              </Tooltip>
            </div>
          </Space>
          <Table
            dataSource={selectedProperties}
            columns={columns}
            scroll={{ y: 510 }}
            className={styles.propertyList}
            data-testid="property-list-table"
            pagination={false}
            expandable={{
              expandIcon: () => <Fragment />,
            }}
          />
        </Space>
      </Modal>
    </Fragment>
  );
}
