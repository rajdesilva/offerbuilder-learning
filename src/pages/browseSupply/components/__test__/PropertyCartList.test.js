import React from "react";
import { createStore } from "redux";
import { render, fireEvent, cleanup } from "../../../../helpers/testUtils";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

import { PropertyCartList } from "../PropertyCartList";
import { tempStore } from "../../../offers/components/__test__/tempStore";
jest.mock("./../../service");
const history = createMemoryHistory();

afterEach(cleanup);
describe("display property component test", () => {
  test("create display property components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const wrapper = render(
      <Router history={history}>
        <PropertyCartList />
      </Router>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });

  test("check all the property details is available or not", () => {
    const store = createStore(() => ({ ...tempStore }));
    const { getByText, getByTestId } = render(
      <Router history={history}>
        <PropertyCartList />
      </Router>,
      {
        store,
      }
    );
    expect(getByText(/Properties in offer/i)).toBeInTheDocument();
    expect(getByTestId("properties-count").textContent).toEqual("1");
  });
  test("check propertyList Button is Disabled or not when data is not there", () => {
    const store = createStore(() => ({ ...tempStore }));
    const { getByText } = render(
      <Router history={history}>
        <PropertyCartList />
      </Router>,
      {
        store,
      }
    );
    const propertyList = getByText(/Properties in offer/i);
    expect(propertyList).toBeEnabled();
  });

  test("check Modals is opening or not after click on property List Button", () => {
    const store = createStore(() => ({ ...tempStore }));
    const { getByTestId, getByText } = render(
      <Router history={history}>
        <PropertyCartList />
      </Router>,
      {
        store,
      }
    );
    const propertyListBtn = getByTestId("property-list-modal");
    fireEvent.click(propertyListBtn);

    // check two buttons
    expect(getByText("Cancel")).toBeInTheDocument();

    expect(getByTestId("proceed").textContent.trim()).toEqual(
      "Proceed to Settings"
    );

    // check for columns
    expect(getByText("Properties")).toBeInTheDocument();
    expect(getByText("Margin")).toBeInTheDocument();
    expect(getByText("Supplier")).toBeInTheDocument();
  });

  test("check cancel Button in the cart modal", () => {
    const store = createStore(() => ({ ...tempStore }));
    const handleProceed = jest.fn();

    const { getByTestId, queryByText } = render(
      <Router history={history}>
        <PropertyCartList
          {...{
            next: handleProceed,
            isEditFlow: false,
          }}
        />
      </Router>,
      {
        store,
      }
    );

    const propertyListBtn = getByTestId("property-list-modal");
    fireEvent.click(propertyListBtn);

    const cancel = getByTestId("cancel");
    fireEvent.click(cancel);
    expect(queryByText("Proceed to Settings")).not.toBeInTheDocument();
  });

  test("check proceed to Button in the cart modal", () => {
    const store = createStore(() => ({ ...tempStore }));

    const proceedFnHandler = jest.fn();

    const { getByTestId, queryByText } = render(
      <Router history={history}>
        <PropertyCartList
          {...{
            next: proceedFnHandler,
            isEditFlow: false,
          }}
        />
      </Router>,
      {
        store,
      }
    );
    const propertyListBtn = getByTestId("property-list-modal");
    fireEvent.click(propertyListBtn);

    const proceedBtn = getByTestId("proceed");
    fireEvent.click(proceedBtn);

    expect(queryByText("Cancel")).not.toBeInTheDocument();
    expect(proceedFnHandler).toHaveBeenCalledTimes(1);
  });

  test("check cart in the Edit Flow ", () => {
    const store = createStore(() => ({ ...tempStore }));

    const proceedFnHandler = jest.fn();

    const { getByTestId, queryByText } = render(
      <Router history={history}>
        <PropertyCartList
          {...{
            next: proceedFnHandler,
            isEditFlow: true,
          }}
        />
      </Router>,
      {
        store,
      }
    );
    const propertyListBtn = getByTestId("property-list-modal");
    fireEvent.click(propertyListBtn);

    expect(queryByText("Cancel")).not.toBeInTheDocument();
    expect(queryByText("Proceed to Settings")).not.toBeInTheDocument();
  });
});
