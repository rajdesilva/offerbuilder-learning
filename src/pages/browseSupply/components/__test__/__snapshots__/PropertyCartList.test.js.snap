// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display property component test create display property components snapshot 1`] = `
<button
  class="ant-btn ant-btn-primary ant-btn-lg ant-btn-background-ghost"
  data-testid="property-list-modal"
  type="button"
>
  <span>
    Properties In Offer
  </span>
  <span>
    Â 
  </span>
  <span
    data-testid="properties-count"
  >
    1
  </span>
</button>
`;
